File: C++_Doc/2_ProgramacionOrientadaObjetos/4_Polimorfismo.cpp
#include <iostream>
using namespace std;

class Inversion {
public:
    virtual int convertirPesoAdolar() = 0;
    virtual int convertirDolarApeso() = 0;
};

class Divisa {
public:
    string tipoMoneda;
    int moneda;
};

class Moneda : public Divisa, public Inversion {
private:
    int peso = 0;
    int dolar = 0;

public:
    void pideDatos() {
        cout << "Ingrese moneda (1: Peso, 2: Dólar):" << endl;
        int opcUsr;
        cin >> opcUsr;
        switch (opcUsr) {
            case 1:
                cout << "Ingrese cantidad en pesos:" << endl;
                cin >> peso;
                tipoMoneda = "Peso";
                break;
            case 2:
                cout << "Ingrese cantidad en dólares:" << endl;
                cin >> dolar;
                tipoMoneda = "Dólar";
                break;
            default:
                cout << "Opción no válida" << endl;
                break;
        }
    }

    int convertirPesoAdolar() override {
        return static_cast<int>(peso / 20.3);
    }

    int convertirDolarApeso() override {
        return static_cast<int>(dolar * 20.3);
    }
};

class Temperatura {
private:
    double fahrenheit = 0;
    double celsius = 0;

public:
    void pideDatos() {
        cout << "Ingrese grados Fahrenheit:" << endl;
        cin >> fahrenheit;
        cout << "Ingrese grados Celsius:" << endl;
        cin >> celsius;
    }

    void muestraTemperatura() {
        cout << "Fahrenheit: " << fahrenheit << endl;
        cout << "Celsius: " << celsius << endl;
    }
};

class Clima {
private:
    Temperatura grado;

public:
    void pideDatos() {
        grado.pideDatos();
    }

    void muestraTemperatura() {
        grado.muestraTemperatura();
    }
};

int main() {
    bool bandera = true;
    
    while (bandera) {
        cout << "Ingrese opción usuario:" << endl;
        cout << "1: Divisas" << endl;
        cout << "2: Temperatura" << endl;
        cout << "3: Salir" << endl;
        int opcUsr;
        cin >> opcUsr;
        
        switch (opcUsr) {
            case 1: {
                Moneda moneda;
                moneda.pideDatos();
                cout << "Peso a dólar: " << moneda.convertirPesoAdolar() << endl;
                cout << "Dólar a peso: " << moneda.convertirDolarApeso() << endl;
                break;
            }
            case 2: {
                Clima clima;
                clima.pideDatos();
                clima.muestraTemperatura();
                break;
            }
            case 3:
                bandera = false;
                break;
            default:
                cout << "Opción no válida" << endl;
                break;
        }
    }
    return 0;
}

File: Java_Doc/src/main/java/org/Alpha/POO/_4_Polimorfismo.java
/*
package org.Alpha.POO;

import java.util.Scanner;

public interface Inversion {
    int convertirPesoAdolar();
    int convertirDolarApeso();
}

public class Divisa {
    String tipoMoneda;
    int moneda;
}

public class Moneda extends Divisa implements Inversion {
    private int peso = 0;
    private int dolar = 0;

    public void pideDatos(Scanner scanner) {
        System.out.println("Ingrese moneda (1: Peso, 2: Dólar):");
        int opcUsr = scanner.nextInt();
        switch (opcUsr) {
            case 1:
                System.out.println("Ingrese cantidad en pesos:");
                this.peso = scanner.nextInt();
                this.tipoMoneda = "Peso";
                break;
            case 2:
                System.out.println("Ingrese cantidad en dólares:");
                this.dolar = scanner.nextInt();
                this.tipoMoneda = "Dólar";
                break;
            default:
                System.out.println("Opción no válida");
                break;
        }
    }

    @Override
    public int convertirPesoAdolar() {
        return (int) (this.peso / 20.3);
    }

    @Override
    public int convertirDolarApeso() {
        return (int) (this.dolar * 20.3);
    }
}

public class Temperatura {
    private double fahrenheit = 0;
    private double celsius = 0;

    public void pideDatos(Scanner scanner) {
        System.out.println("Ingrese grados Fahrenheit:");
        this.fahrenheit = scanner.nextDouble();
        System.out.println("Ingrese grados Celsius:");
        this.celsius = scanner.nextDouble();
    }

    public void muestraTemperatura() {
        System.out.println("Fahrenheit: " + this.fahrenheit);
        System.out.println("Celsius: " + this.celsius);
    }
}

public class Clima {
    private Temperatura grado = new Temperatura();

    public void pideDatos(Scanner scanner) {
        grado.pideDatos(scanner);
    }

    public void muestraTemperatura() {
        grado.muestraTemperatura();
    }
}

public class _4_Polimorfismo {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        boolean bandera = true;

        while (bandera) {
            System.out.println("Ingrese opción usuario:");
            System.out.println("1: Divisas");
            System.out.println("2: Temperatura");
            System.out.println("3: Salir");
            int opcUsr = scanner.nextInt();

            switch (opcUsr) {
                case 1:
                    Moneda moneda = new Moneda();
                    moneda.pideDatos(scanner);
                    System.out.println("Peso a dólar: " + moneda.convertirPesoAdolar());
                    System.out.println("Dólar a peso: " + moneda.convertirDolarApeso());
                    break;
                case 2:
                    Clima clima = new Clima();
                    clima.pideDatos(scanner);
                    clima.muestraTemperatura();
                    break;
                case 3:
                    bandera = false;
                    break;
                default:
                    System.out.println("Opción no válida");
                    break;
            }
        }
        scanner.close();
    }
}
*/
